Unosquare.FFME.Windows/Rendering/D3DVideoRenderer.cs

```diff
@@ -0,0 +1,417 @@
+                        ? CreateFlags.HardwareVertexProcessing
+                        : CreateFlags.SoftwareVertexProcessing;
+
+                    DeviceCreationFlags = CreateFlags.Multithreaded | CreateFlags.FpuPreserve | vertexMode;
+                }
+                catch
+                {
+                    IsD3DAvailable = false;
+                }
+            }
+
+            public GraphicsBuffer(bool useBackBuffer)
+            {
+                HasBackBuffer = useBackBuffer;
+            }
+
+            /// <summary>
+            /// Gets a value indicating whether the D3DEx Api is available.
+            /// </summary>
+            public static bool IsD3DAvailable { get; }
+
+            public bool HasBackBuffer { get; }
+
+            public bool HasNewPicture { get; private set; }
+
+            /// <summary>
+            /// Gets the D3D engine.
+            /// </summary>
+            private static Direct3DEx Engine { get; }
+
+            /// <summary>
+            /// Gets the D3D device creation flags.
+            /// </summary>
+            private static CreateFlags DeviceCreationFlags { get; }
+
+            /// <summary>
+            /// Gets the device presentation parameters.
+            /// </summary>
+            private static PresentParameters DeviceParameters { get; } = new PresentParameters
+            {
+                Windowed = true,
+                SwapEffect = SwapEffect.Discard,
+                PresentationInterval = PresentInterval.Default,
+                BackBufferFormat = Format.Unknown,
+                BackBufferWidth = 1,
+                BackBufferHeight = 1,
+            };
+
+            /// <summary>
+            /// Gets a valid D3D device.
+            /// </summary>
+            private DeviceEx Device
+            {
+                get
+                {
+                    EnsureDeviceAvailable();
+                    return m_Device;
+                }
+            }
+
+            public void Clear(D3DImage image)
+            {
+                lock (SyncLock)
+                {
+                    if (image == null || !image.IsFrontBufferAvailable)
+                        return;
+
+                    image.Lock();
+                    image.SetBackBuffer(D3DResourceType.IDirect3DSurface9, IntPtr.Zero);
+                    image.Unlock();
+                }
+            }
+
+            public void Write(VideoBlock block)
+            {
+                lock (SyncLock)
+                {
+                    if (IsDsiposing)
+                        return;
+
+                    if (HasBackBuffer)
+                        WriteToBackBuffer(block);
+                    else
+                        WriteToFrontBuffer(block);
+
+                    HasNewPicture = true;
+                }
+            }
+
+            public bool Render(D3DImage image, TimeSpan renderTime)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change return type to 'void'; not a single caller uses the returned value.](https://app.codacy.com/app/UnosquareLabs/ffmediaelement/pullRequest?prid=4677725)